server/plugin.go

```diff
@@ -582,7 +645,15 @@ func (p *Plugin) HasProjectHook(ctx context.Context, user *gitlab.UserInfo, name
 // HasGroupHook checks if the subscribed GitLab Group has a webhook
 // with a URL that matches the Mattermost Site URL.
 func (p *Plugin) HasGroupHook(ctx context.Context, user *gitlab.UserInfo, namespace string) (bool, error) {
-	hooks, err := p.GitlabClient.GetGroupHooks(ctx, user, namespace)
+	var hooks []*gitlab.WebhookInfo
+	err := p.useGitlabClient(user, func(info *gitlab.UserInfo, token *oauth2.Token) error {
```

Comment:
Since `useGitlabClient` doesn't actually provide a client, maybe we should name it `useGitlabToken`?

Reply:
It could be either way. The name made sense to me since we're using it to wrap the calls to `p.GitlabClient` methods.